### PR TITLE
Update spec's title

### LINKED ISSUE #2030: Show "Preview of #" in title
See the following for details:
https://github.com/tobie/pr-preview/issues/23

Basically, the spec's title should come "Preview of PR xxx: " when the spec `isPreview` is true. 

This would build on @yatri1609's work: 
https://github.com/w3c/respec/pull/1843

### PR DESCRIPTION
**Pull request by yatri1609**
closes #2030 

This PR renames the title to `Preview of PR xxx` from "This is a preview of pull request `PR number`".

@sidvishnoi @marcoscaceres Please have a look at it.

### CODE REVIEW

**Comment by yatri1609** [comment]
@marcoscaceres Sure I'll do that and submit it

**Comment by yatri1609** [comment]
@marcoscaceres Please have a look at it

**Comment by marcoscaceres** [comment]
@yatri1609 thanks! having a look now.

File: src/w3c/templates/cgbg-headers.js [hunk 1]
```diff
@@ -9,11 +9,20 @@ export default conf => {
     <div class="head">
       ${conf.logos.map(showLogo)}
       <h1 class="title p-name" id="title">${conf.title}</h1>
-      ${conf.subtitle
-        ? html`
-            <h2 id="subtitle">${conf.subtitle}</h2>
-          `
-        : ""}
+      ${
```
Comment (marcoscaceres) [inline src/w3c/templates/cgbg-headers.js]: I'm not sure we need this block, as we will end up with two `h1`s (see line 11). Also, it seems we can indeed just rely on `conf.title`.

Comment (yatri1609) [reply src/w3c/templates/cgbg-headers.js]: We can just remove the `<h1 class="title p-name" id="title">${conf.title}</h1>` from the code, I guess

Comment (marcoscaceres) [reply src/w3c/templates/cgbg-headers.js]: No, I think we want to keep that first one. All specs need to have a `<h1 class="title...">` so it makes sense just to leave it be.

Comment (yatri1609) [reply src/w3c/templates/cgbg-headers.js]: @marcoscaceres We can just go with Line 11 only, I guess and not add the existing block

Comment (yatri1609) [reply src/w3c/templates/cgbg-headers.js]: I'll send in the updates code.

File: src/w3c/templates/cgbg-sotd.js [hunk 2]
```diff
@@ -4,32 +4,37 @@ export default (conf, opts) => {
...
-                    <a href="${conf.edDraftURI}">${conf.edDraftURI}</a> for the
-                    Editor's draft.
-                  `
-                : ""}
-            </p>
-          </details>
-        `
-      : ""}
+    ${
+      conf.isPreview
```
Comment (marcoscaceres) [inline src/w3c/templates/cgbg-sotd.js]: @saschanaz do you recall why ESLint is doing this with the whitespace? I can recall how to fix this :( (Visual Studio Code seems to do the right thing)

Comment (marcoscaceres) [reply src/w3c/templates/cgbg-sotd.js]: can't recall*

Comment (saschanaz) [reply src/w3c/templates/cgbg-sotd.js]: prettier/prettier#5771 introduced this behavior. Maybe we can `npm run lint -- --fix` again and things will be fixed.

File: src/w3c/templates/sotd.js [hunk 3]
```diff
@@ -29,6 +28,35 @@ export default (conf, opts) => {
           </details>
         `
       : ""}
+    ${conf.isUnofficial
```
Comment (marcoscaceres) [inline src/w3c/templates/sotd.js]: Not sure why this change is here?

Comment (saschanaz) [reply src/w3c/templates/sotd.js]: Looking at the commit history, I think this is accidentally added when merging conflicts.

Comment (yatri1609) [reply src/w3c/templates/sotd.js]: Should I remove this? As this was a part of the merge which I did earlier

Comment (saschanaz) [reply src/w3c/templates/sotd.js]: This part had been existed, modified after you opened this PR, and then deleted, which is why this appeared when merging. Should be removed anyway.

Comment (yatri1609) [reply src/w3c/templates/sotd.js]: Ohk I'll remove it

**Comment by saschanaz** [comment]
Should we write a test for this, as it seems nothing tests the title?

**Comment by marcoscaceres** [comment]
Yes, absolutely. The spec title still needs to be updated somehow.

File: src/w3c/templates/cgbg-sotd.js [hunk 4]
```diff
@@ -7,15 +7,14 @@ export default (conf, opts) => {
     ${conf.isPreview
       ? html`
           <details class="annoying-warning" open="">
-            <summary
-              >This is a
-              preview${conf.prUrl && conf.prNumber
+            <title>
```
Comment (marcoscaceres) [inline src/w3c/templates/cgbg-sotd.js]: Not quite. You need to find where ReSpec creates the `h1` title of the spec AND find where the title element is added and then either replace it or update it there.

**Comment by marcoscaceres** [comment]
Closing as per discussion.
